go/tasks/plugins/array/k8s/transformer.go

```diff
@@ -99,6 +137,22 @@ func FlyteArrayJobToK8sPodTemplate(ctx context.Context, tCtx core.TaskExecutionC
 			Annotations:     annotations,
 			OwnerReferences: []metav1.OwnerReference{tCtx.TaskExecutionMetadata().GetOwnerReference()},
 		},
-		Spec: *podSpec,
-	}, arrayJob, nil
+	}
+	if taskTemplate.GetContainer() != nil {
+		podSpec, err := flytek8s.ToK8sPodSpec(ctx, arrTCtx)
+		if err != nil {
+			return v1.Pod{}, nil, err
+		}
+		pod.Spec = *podSpec
+	} else if taskTemplate.GetK8SPod() != nil {
+		k8sPod, err := buildPodMapTask(taskTemplate, tCtx.TaskExecutionMetadata())
```

Comment:
How does this resolve command line args in the primary container?
`pyflyte-execute --inputsFile {{.inputs}}`
where does that get resolved? it's not clear to me...

Reply:
done here: https://github.com/flyteorg/flyteplugins/pull/182/files#diff-01ee392f44bbd91e1e5f4afd999327e02f8f1df54fd5b5b576478e903cbcb3c2R113

Reply:
Aha, I guess we should move https://github.com/flyteorg/flyteplugins/pull/182/files#diff-f6e976bede1900c30812a8f3dc9c733a2a60381b4ad8e38fe33f0525bf0bb118R113-R116 into the `if container` condition above?

Reply:
@EngHabu not sure - do we allow for input substitution with the custom?

Reply:
updated @EngHabu